bitset.h

```diff
@@ -0,0 +1,156 @@
+#ifndef __BITSET_H__
+#define __BITSET_H__
+#include <stdint.h>
+#include <stddef.h>
+#include "common.h"
+
+/* Bit operations on simple values. */
+#define BIT_SET(n, b) ((n) |=  (1 << (b)) )
+#define BIT_CLR(n, b) ((n) &= ~(1 << (b)) )
+#define BIT_INV(n, b) ((n) ^=  (1 << (b)) )
+#define BIT_GET(n, b) (((n) & (1<<(b))) != 0)
+
+/* TODO: This should be in some other file. */
+
+/* ============================== */
+/*   Bitsets                      */
+
+/* Defines a variable of bitset<size> type, named "name". This can be
+ * used both for creating a bitset variable, as well as for defining a
+ * bitset field within a struct. */
+#define BITSET_DEFINE(name, size)                       \
+    struct name ## _bitset ## size ##_t {               \
+        WORD_TYPE bf[_BITSET_WORDS_NO(size)];           \
+    } name;
+
+/* Macros for initialising bitsets. */
+#define BITSET_CLEAR(bitset,size)                       \
+    do{                                                 \
+        size_t i;                                       \
+        for( i = 0; i < _BITSET_WORDS_NO(size); i++)    \
+            bitset.bf[i] = 0;                           \
+    } while(0)
+#define BITSET_CLEAR(bitset,size)                       \
+    do{                                                 \
+        size_t i;                                       \
+        for( i = 0; i < _BITSET_WORDS_NO(size); i++)    \
+            bitset.bf[i] = 0;                           \
+    } while(0)
+
+/* Bit operations on a bitset. */
+#define BITSET_BIT_SET(bitset, bitno)                   \
+    bitset.bf[_BITSET_BIT_WORD(bitno)] |=               \
+        _BITSET_BIT_MASK(bitno)
+
+#define BITSET_BIT_CLR(bitset, bitno)                   \
+    bitset.bf[_BITSET_BIT_WORD(bitno)] &=               \
+       ~_BITSET_BIT_MASK(bitno)
+
+#define BITSET_BIT_INV(bitset, bitno)                   \
+    bitset.bf[_BITSET_BIT_WORD(bitno)] ^=               \
+        _BITSET_BIT_MASK(bitno)
+
+/* Returns the value of the queried bit in a bitset. */
+#define BITSET_GET(bitset, bitno)                       \
```

Comment:
Do you mean that I should rename all `BITSET_*` to `BITMAP_*`, or just `BITSET_BIT_*` ?